### PR TITLE
cmd/puppeth: your Ethereum private network manager

### PR DESCRIPTION
**Pull request by karalabe**
Puppeth

### CODE REVIEW

File: cmd/faucet/faucet.go [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+			} `json:"files"`
+		}
+		err = json.NewDecoder(res.Body).Decode(&gist)
+		res.Body.Close()
+		if err != nil {
+			websocket.JSON.Send(conn, map[string]string{"error": err.Error()})
+			continue
+		}
+		if gist.Owner.Login == "" {
+			websocket.JSON.Send(conn, map[string]string{"error": "Nice try ;)"})
```
Comment (fjl) [inline cmd/faucet/faucet.go]: It's usually best to split up code into functions so the handling of the error can be done once. If you move handling of gists and their content into a function, you could remove most of the

```go
websocket.JSON.Send(conn, map[string]string{"error": "..."})
continue
```

File: cmd/faucet/faucet.go [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+			URL string `json:"url"`
+		}
+		if err := websocket.JSON.Receive(conn, &msg); err != nil {
+			return
+		}
+		if !strings.HasPrefix(msg.URL, "https://gist.github.com/") {
+			websocket.JSON.Send(conn, map[string]string{"error": "URL doesn't link to GitHub Gists"})
+			continue
+		}
+		log.Info("Faucet funs requested", "gist", msg.URL)
```
Comment (fjl) [inline cmd/faucet/faucet.go]: funs -> funds

File: cmd/faucet/faucet.html [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    <div class="vertical-center">
+      <div class="container">
+  			<div class="row" style="margin-bottom: 16px;">
+          <div class="col-lg-12">
+            <h1 style="text-align: center;"><i class="fa fa-bath" aria-hidden="true"></i> {{.Network}} GitHub Authenticated Faucet <i class="fa fa-github-alt" aria-hidden="true"></i></h1>
+          </div>
+        </div>
+  			<div class="row">
+					<div class="col-lg-8 col-lg-offset-2">
+						<div class="input-group">
```
Comment (fjl) [inline cmd/faucet/faucet.html]: HTML indentation is totally nuts here. Maybe use a formatter.

**Comment by karalabe** [comment]
@fjl PTAL. I fixed the typo and the indentation. I did not have time tonight to address the error handling. Please consider whether you can live with the current code, and if not, I'll try to address it tomorrow.

**Comment by fjl** [comment]
I can live with almost anything. Would still be nice if you fixed it later today.

**Pull request merged**